sycl/doc/design/SYCLBINDesign.md

```diff
@@ -0,0 +1,284 @@
+
+An native device code image contains the binary data for the corresponding
+module AOT compiled for a specific device, identified by the architecture
+string.
+
+
+##### Native device code image header
+
+A native device code image header contains the following fields in the stated
+order:
+
+| Type       | Description                                                                         |
+| ---------- | ----------------------------------------------------------------------------------- |
+| `uint64_t` | Byte offset of the metadata from the start of the metadata byte table.              |
+| `uint64_t` | Byte size of the metadata in the metadata byte table.                               |
+| `uint64_t` | Byte offset of the device code image bytes from the start of the binary byte table. |
+| `uint64_t` | Byte size of the device code image bytes in the binary byte table.                  |
+
+
+##### Native device code image metadata
+
+A native device code image metadata entry contains a single property set with
+the identifying name "SYCLBIN/native device code image module metadata", as
+described in the
+[PropertySets.md](PropertySets.md#syclbinnative-device-code-image-metadata)
+design document.
+
+
+### Byte tables
+
+A byte table contains dynamic data, such as metadata and binary blobs. The
+contents of it is generally referenced by an offset specified in the headers.
+
+
+### SYCLBIN version changelog
+
+The SYCLBIN format is subject to change, but any such changes must come with an
+increment to the version number in the header and a subsection to this section
+describing the change.
+
+Additionally, any changes to the property set structure that affects the way the
+runtime has to parse the contained property sets will require an increase in the
+SYCLBIN version. Adding new property set names or new predefined properties only
+require a SYCLBIN version change if the the SYCLBIN consumer cannot safely
+ignore the property.
+
+
+#### Version 1
+
+ * Initial version of the layout.
+
+
+## Clang driver changes
+
+The clang driver needs to accept the following new flags:
+
+<table>
+<tr>
+<th>Option</th>
+<th>Description</th>
+</tr>
+<tr>
+<td>`-fsyclbin`</td>
+<td>
+If this option is set, the output of the invocation is a SYCLBIN file with the
+.syclbin file extension. This skips the host-compilation invocation of the typical
+`-fsycl` pipeline, instead passing the output of the clang-offload-packager
+invocation to clang-linker-wrapper together with the new `--syclbin` flag.
+
+Setting this option will override `-fsycl` and `-fsycl-device-only`.
+
+The behavior is dependent on using the clang-linker-wrapper. As the current default
+offload compilation behavior is using the old offload model (driver based), this
+option currently requires `--offload-new-driver` to be set.
+</td>
+</tr>
+<tr>
+<td>`--offload-ir`</td>
+<td>*TODO*</td>
+</tr>
+<tr>
+<td>`--offload-rdc`</td>
+<td>This is an alias of `-fgpu-rdc`.</td>
```

Comment:
Not much information in regards to these options and their purpose in the context of SYCLBIN.  Can these be embellished or just added later?

Reply:
I think `--offload-rdc` is just intended to be an alias for `-fgpu-rdc`, mainly to make the option more general. Maybe it doesn't really need to be part of this document? I think it's mostly just bikeshedding in the end.

I will remove `--offload-ir` until I know how it will actually be used. 🫡